### PR TITLE
Use disused:shop instead of shop=vacant

### LINKED ISSUE #2707: Use disused:shop instead of shop=vacant
If a shop is closed StreetComplete removes all tags and adds [`shop=vacant`](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dvacant). This tag has become rather controversial, though. This place is no longer a shop, so it shouldn't get tagged as a shop. Also, every data consumer needs to add a special handling for `shop=vacant`, i.e. for excluding it when rendering shops. Therefore it would be better to use the [lifecycle prefix](https://wiki.openstreetmap.org/wiki/Lifecycle_prefix) instead, i.e. `disused:shop=<former shop value>`.

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #2707

And yes, it is (now) properly tested.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/ShopGoneDialog.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ class ShopGoneDialog(
         getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
             when (selectedRadioButtonId) {
                 R.id.vacantRadioButton -> {
-                    onSelectedFeature(mapOf("shop" to "vacant"))
+                    onSelectedFeature(mapOf("disused:shop" to "yes"))
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/ShopGoneDialog.kt:71]: Reading the code we had:
`shop=supermarket` => `shop=vacant`
and now we have:
`shop=supermarket` => unchanged
**added** `disused:shop=yes`

I think we should do
`shop=supermarket` => `disused:shop=yes`

Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/ShopGoneDialog.kt:71]: definitely, not sure how I managed to miss it

Comment (matkoniecz) [reply app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/ShopGoneDialog.kt:71]: Or maybe it actually works? Looking again at code it works the same way as other feature replacement (`onSelectedFeature(feature.addTags)`)

Yeah, it works.

`D/QuestController: Replaced NODE #6325607440 in frame of quest AddOpeningHours with DELETE "shop"="gift", DELETE "noname"="yes", ADD "disused:shop"="yes"`

**Comment by westnordost** [comment]
Yes, it works because it ultimatively calls `QuestController::createReplaceShopChanges` which removes **ALL** tags from that element except these:
```kotlin
private val KEYS_THAT_SHOULD_NOT_BE_REMOVED_WHEN_SHOP_IS_REPLACED = listOf(
    "landuse", "historic",
    // building/simple 3d building mapping
    "building", "man_made", "building:.*", "roof:.*",
    // any address
    "addr:.*",
    // shop can at the same time be an outline in indoor mapping
    "level", "level:ref", "indoor", "room",
    // geometry
    "layer", "ele", "height", "area", "is_in",
    // notes and fixmes
    "FIXME", "fixme", "note"
).map { it.toRegex() }
```

**Review by westnordost — APPROVED** [review]

**Comment by westnordost** [comment]
By the way, you chose to tag `disused:shop=yes` and not `disused:$previousKey=$previousValue`. Was this deliberate?

**Comment by matkoniecz** [comment]
> By the way, you chose to tag disused:shop=yes and not disused:$previousKey=$previousValue. Was this deliberate?

`$previousValue` is deliberately ignored. See https://github.com/streetcomplete/StreetComplete/issues/2707#issuecomment-811265793 https://github.com/streetcomplete/StreetComplete/issues/2707#issuecomment-816471983

`$previousKey` - it is always shop, right? (and if not then it is not worse anyway as it replaces `shop=vacant`)

**Comment by westnordost** [comment]
Ok, just asking. 

No in some cases it may be an amenity. For example `amenity=pharmacy`

**Comment by FloEdelmann** [comment]
@westnordost There is a typo in the changelog for this PR:

```diff
- Vacant shops are not tagged as disused:shop=*
+ Vacant shops are now tagged as disused:shop=*
```

I'm currently not at a computer, so I can't open a PR directly.

**Comment by westnordost** [comment]
oh damn, the new version is released already

**Comment by riQQ** [comment]
The GitHub release also has the typo

**Pull request merged**